web/src/pages/billing/SequencingGroupsByMonth.tsx

```diff
@@ -0,0 +1,263 @@
+import { debounce } from 'lodash'
+import * as React from 'react'
+import { useRef } from 'react'
+import { useNavigate, useParams } from 'react-router-dom'
+import { useMeasure } from 'react-use'
+import { Button, Card, Grid, Message } from 'semantic-ui-react'
+import LoadingDucks from '../../shared/components/LoadingDucks/LoadingDucks'
+import ProjectSelector, { IMetamistProject } from '../project/ProjectSelector'
+
+import * as Plot from '@observablehq/plot'
+import { PaddedPage } from '../../shared/components/Layout/PaddedPage'
+import { SequencingGroupApi } from '../../sm-api'
+import './components/BillingCostByTimeTable.css'
+
+type ProjectHistory = {
+    [key: string]: {
+        [key: string]: number
+    }
+}
+
+interface TypeData {
+    date: Date
+    type: string
+    count: number
+}
+
+/* eslint-disable @typescript-eslint/no-explicit-any  -- too many anys in the file to fix right now but would be good to sort out when we can */
+const SequencingGroupsByMonth: React.FunctionComponent = () => {
+    // Use navigate and update url params
+    const navigate = useNavigate()
+    const { projectName } = useParams()
+
+    // Data loading
+    const [isLoading, setIsLoading] = React.useState<boolean>(true)
+    const [error, setError] = React.useState<string | undefined>()
+    const [message, setMessage] = React.useState<string | undefined>()
+    const [data, setData] = React.useState<TypeData[]>([])
+
+    // Chart refs
+    const containerRef = useRef<HTMLDivElement>(null)
+    const [measureRef, { width, height }] = useMeasure<HTMLDivElement>()
+
+    // Callback to get data for plotting.
+    const onProjectSelect = (_project: IMetamistProject) => {
+        navigate(`/billing/sequencingGroupsByMonth/${_project.name}`)
+    }
+
+    const getData = React.useCallback(async (project: string | undefined) => {
```

Comment:
Once this is switched to the graphql API it will make data fetching a fair bit simpler. The hooks that apollo-graphql provides mean that you don't need to use effects or callbacks for data fetching. If you search for `useQuery` in the code you should find plenty of cases where graphql fetches happen.